columnflow/ml/__init__.py

```diff
@@ -185,6 +195,26 @@ def _format_value(self: MLModel, value: Any) -> str:
         # any other case
         return str(value)
 
+    @property
+    def parameters_repr(self: MLModel) -> str:
+        """
+        Returns a hash of string representation of all parameters. This is used to uniquely identify
+        a model instance based on its parameters.
+
+        :raises: Exception in case the parameters_repr changed after it was set.
+        :returns: String representation of all parameters.
+        """
+        if not self.parameters:
+            return ""
+        parameters_repr = law.util.create_hash(self._join_parameter_pairs(only_significant=True))
+        if hasattr(self, "_parameters_repr") and self._parameters_repr != parameters_repr:
+            raise Exception(
+                f"parameters_repr changed from {self._parameters_repr} to {parameters_repr};"
+                "this should not happen",
+            )
```

Comment:
I lik ethe mechanism to check for potential changes in the architecture. However, is this not sensitive to permutation within the `ml-model-settings` parameter? E.g., `--ml-model-settings "batchsize=2048,layers=32;32;32"` and `--ml-model-settings "layers=32;32;32,batchsize=2048"` should yield the same architecture, but I think the hashes would be different, right? Have you tested this by chance? Perhaps it's better to first sort the tuple entries

Reply:
Hi @pkausw, I agree that the order should not matter, however, the `self.parameters` has explicitly been initialized as an OrderedDict. Maybe @riga can give a comment whether there is a reason to order the parameters. Otherwise, I'd also be in favour of sorting the parameters before hashing (which I am already doing in my own analysis)

Reply:
I agree, this seems to be the safer thing to do. The change is also minimal, so I guess it should be easy to change this if necessary 👍